src/integrations/windowsIntegrationManager.ts

```diff
@@ -1,14 +1,428 @@
+   * Output is logged to the log file.
+   */
+  protected async execCommand(opts: {distro?: string, encoding?:BufferEncoding, root?: boolean, env?: Record<string, string>}, ...command: string[]):Promise<void> {
+    const logStream = opts.distro ? Logging[`wsl-helper.${ opts.distro }`] : console;
+    const args = [];
+
+    if (opts.distro) {
+      args.push('--distribution', opts.distro);
+      if (opts.root) {
+        args.push('--user', 'root');
+      }
+      args.push('--exec');
+    }
+    args.push(...command);
+    console.debug(`Running ${ await this.wslExe } ${ args.join(' ') }`);
+
+    await spawnFile(
+      await this.wslExe,
+      args,
+      {
+        env:         opts.env,
+        encoding:    opts.encoding ?? 'utf-8',
+        stdio:       ['ignore', logStream, logStream],
+        windowsHide: true,
+      }
+    );
+  }
+
+  protected async captureCommand(opts: {distro?: string, encoding?: BufferEncoding, env?: Record<string, string>}, ...command: string[]):Promise<string> {
+    const logStream = opts.distro ? Logging[`wsl-helper.${ opts.distro }`] : console;
+    const args = [];
+
+    if (opts.distro) {
+      args.push('--distribution', opts.distro, '--exec');
+    }
+    args.push(...command);
+    console.debug(`Running ${ await this.wslExe } ${ args.join(' ') }`);
+
+    const { stdout } = await spawnFile(
+      await this.wslExe,
+      args,
+      {
+        env:         opts.env,
+        encoding:    opts.encoding ?? 'utf-8',
+        stdio:       ['ignore', 'pipe', logStream],
+        windowsHide: true,
+      }
+    );
+
+    return stdout;
+  }
+
+  /**
+   * Return the Linux path to the WSL helper executable.
+   */
+  protected async getLinuxToolPath(distro: string, ...tool: string[]): Promise<string> {
+    // We need to get the Linux path to our helper executable; it is easier to
+    // just get WSL to do the transformation for us.
+
+    const logStream = Logging[`wsl-helper.${ distro }`];
+    const { stdout } = await spawnFile(
+      await this.wslExe,
+      ['--distribution', distro, '--exec', '/bin/wslpath', '-a', '-u',
+        path.join(paths.resources, 'linux', ...tool)],
+      { stdio: ['ignore', 'pipe', logStream] }
+    );
+
+    return stdout.trim();
+  }
+
+  protected async syncSocketProxy(): Promise<void> {
+    const shouldRun =
+      this.enforcing &&
+      this.backendReady &&
+      this.settings.kubernetes?.containerEngine === ContainerEngine.MOBY;
+
+    console.debug(`Syncing socket proxy: ${ shouldRun ? 'should' : 'should not' } run.`);
+    if (shouldRun) {
+      this.windowsSocketProxyProcess.start();
+    } else {
+      this.windowsSocketProxyProcess.stop();
+    }
+    await Promise.all((await this.distros).map(distro => this.syncDistroSocketProxy(distro, shouldRun)));
+  }
+
+  /**
+   * SyncDistroProcessState ensures that the background process for the given
+   * distribution is started or stopped, as desired.
+   * @param distro The distribution to manage.
```

Comment:
`shoiuldRun` not described in jsdoc

Reply:
Fixed in 095f551c5065c3551b5cfd9719569dea95d4240b